## TRAVAIL SUR EAU DU ROBINET 

In [1]:
import pandas as pd

pd.set_option("display.max_columns", None)  # show all cols
pd.set_option("display.max_colwidth", None)  # show full width of showing cols
pd.set_option("display.expand_frame_repr", False)

In [2]:
edr_ttp_plv = pd.read_csv(
    r"C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\TTP_PLV_202412.txt", delimiter=","
)
edr_ttp_plv.head(2)

distrcode                                                   distrlib      distrsiret cddept inseecommune         nomcommune   cdreseau                           nomreseau codetypeinstallation               nomtypeinstallation inae    cdpointsurv                       nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
0  00101028  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  25010183900059    001        01004  AMBERIEU-EN-BUGEY  001000245  TTP (CLG) AMBERIEU VILLE VAREILLES                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0010000000252  TTP (CLG) AMBERIEU VILLE VAREILLE   00100143348  2024-11-25     09h02           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00100452  P1                 INF  13000807100107  250101839                        C                        C             1             NaN           NaN       NaN      NaN  2024-12-17  2024-12-18  2025-01-06                O      NaN      NaN
1  00101028  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  25010183900059    001        01004  AMBERIEU-EN-BUGEY  001000246       TTP (CLG) AMBERIEU ST GERMAIN                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0010000000253      TTP (CLG) AMBERIEU ST GERMAIN   00100143573  2024-12-26     08h37           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00100452  P1                 NaN  13000807100107  250101839                        C                        C             1             NaN           NaN       NaN      NaN  2024-12-31  2025-01-01  2025-01-06                O      NaN      NaN

In [3]:
edr_ttp_plv.describe(include="all")

distrcode distrlib      distrsiret cddept inseecommune nomcommune   cdreseau nomreseau codetypeinstallation               nomtypeinstallation  inae    cdpointsurv nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                         libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
count       6593     6593            4199   6593         6593       6593       6593      6593                 6593                              6593  6593           6593         6593          6593        6593      6589         6593                                                                                                     6593      6593                               6593                  6593                  6593       6593                 6557       6593          981            3228       4199                     6593                     6593   6593.000000             0.0           0.0       0.0      0.0        6593        6593        6593             6593      0.0      0.0
unique      1712     1603             448     99         4207       4162       4867      4777                    1                                 1     3           4999         4306          6593         132       499            7                                                                                                     1170         9                                  9                     3                     4          1                 1761         34           11              49        358                        3                        3           NaN             NaN           NaN       NaN      NaN          44          43        1327                1      NaN      NaN
top     01700013     SAUR  57202552610945    062        75056      PARIS  038000306  ROMANCHE                  TTP  STATION DE TRAITEMENT-PRODUCTION   ESO  0380000000385    RESERVOIR   97600032305  2024-11-26     09h30           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T1  ESO A TURB. < 2 SORTIE PRODUCTION                     C                     C          O             01700013  P1                 INF  41054531300018  339379984                        C                        C           NaN             NaN           NaN       NaN      NaN  2024-12-16  2024-12-17  2023-11-22                O      NaN      NaN
freq          82      193             548    208           42         42         29        29                 6593                              6593  5217             29          113             1         280        90         5932                                                                                                     3791      2841                               2841                  5361                  5994       6593                   82       4209          474             564        830                     5155                     4986           NaN             NaN           NaN       NaN      NaN         428         427          67             6593      NaN      NaN
mean         NaN      NaN             NaN    NaN          NaN        NaN        NaN       NaN                  NaN                               NaN   NaN            NaN          NaN           NaN         NaN       NaN          NaN                                                                                                      NaN       NaN                                NaN                   NaN                   NaN        NaN                  NaN        NaN          NaN             NaN        NaN                      NaN         

In [4]:
edr_ttp_res = pd.read_csv(
    r"C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\TTP_RES_202412.txt", delimiter=","
)
edr_ttp_res.head(2)

C:\Users\EPPINGER Muriel\AppData\Local\Temp\ipykernel_15652\845498637.py:1: DtypeWarning: Columns (0,1,2,5,8) have mixed types. Specify dtype option on import or set low_memory=False.
  edr_ttp_res = pd.read_csv(r'C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\TTP_RES_202412.txt', delimiter=',')


cddept referenceprel    cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux rqana rssigne  rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
0      1     100142348  100151329.0      LSE2409-52850          2024-11-20        6901.0  4.105453e+13        0011AU           101003.0  2024-11-26         L       2060.0             ACTRN22  2,70     NaN    2.7                     Bq/L                9                NaN              217.0                O
1      1     100142348  100151329.0      LSE2409-52850          2024-11-20        6901.0  4.105453e+13        0011AU           101003.0  2024-11-26         L       6489.0                 ASP     0     NaN    0.0               SANS OBJET                X                NaN              217.0                O

In [5]:
edr_ttp_res.describe(include="all")

cddept  referenceprel     cdanalyse refechantillonlabo datereceptionechant  cdlaboratoire   laboratoire cdtypeanalyse  analysecdorganisme analysedate insituana    cdparametre cdparametresiseeaux   rqana rssigne          rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
count   680081.0   6.800810e+05  6.450860e+05             645083              642795       645086.0  6.450830e+05        642752            645064.0      644267    680081  679211.000000              678260  680016  537916  677964.000000                   680081           680081          35.000000       5.558500e+05           680068
unique     101.0   6.595000e+03  6.840000e+03               6822                 130           54.0           NaN           728              1795.0         161         2            NaN                1237    7836       7            NaN                       36               36                NaN                NaN                1
top         59.0   7.500210e+09  7.500210e+09     2024.27065-1-1          2024-11-25         6901.0           NaN       037P1P2           1700013.0  2024-12-13         L            NaN                  PH  <0,005       <            NaN                     µg/L              133                NaN                NaN                O
freq     19730.0   6.080000e+02  6.040000e+02                682               29078       250802.0           NaN         13820             13350.0       24269    645086            NaN                7564  193365  535382            NaN                   504300           504300                NaN                NaN           680068
mean         NaN            NaN           NaN                NaN                 NaN            NaN  3.322055e+13           NaN                 NaN         NaN       NaN    2620.384811                 NaN     NaN     NaN       6.725650                      NaN              NaN          22.885714       2.618557e+12              NaN
std          NaN            NaN           NaN                NaN                 NaN            NaN  1.684008e+13           NaN                 NaN         NaN       NaN    2432.925613                 NaN     NaN     NaN      78.484704                      NaN              NaN           0.676123       5.837646e+12              NaN
min          NaN            NaN           NaN                NaN                 NaN            NaN  1.300021e+13           NaN                 NaN         NaN       NaN       0.000000                 NaN     NaN     NaN     -29.200000                      NaN              NaN          19.000000       9.800000e+01              NaN
25%          NaN            NaN           NaN                NaN                 NaN            NaN  2.000132e+13           NaN                 NaN         NaN       NaN    1303.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN          23.000000       2.100000e+02              NaN
50%          NaN            NaN           NaN                NaN                 NaN            NaN  4.105453e+13           NaN                 NaN         NaN       NaN    1676.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN          23.000000       2.170000e+02              NaN
75%          NaN            NaN           NaN                NaN                 NaN            NaN  4.105453e+13           NaN                 NaN         NaN       NaN    2738.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN          23.000000       2.270000e+02              NaN
max          NaN            NaN           NaN                NaN                 NaN            NaN  7.788520e+13           NaN                 NaN         NaN       NaN   99999.000000                 NaN     NaN     NaN   39000.000000                      NaN              NaN          23.000000       2.000132e+13              NaN

In [6]:
# modification format reference prel pour comparaison avec autres bdd

edr_ttp_res["referenceprel"] = edr_ttp_res["referenceprel"].astype("string")
edr_ttp_res["referenceprel"] = edr_ttp_res["referenceprel"].apply(
    lambda x: "00" + x if len(x) == 9 else "0" + x if len(x) == 10 else x
)

Vérification faite le prélèvement '00100143348' n'est pas présent dans la base EDR_UDI

In [7]:
# recherche de doublons dans la base des prélevements

doublons = edr_ttp_plv.duplicated()
df_doublons_plv = edr_ttp_plv[doublons]
print(df_doublons_plv)

# recherche de doublons dans la base des résultats

doublons = edr_ttp_res.duplicated()
df_doublons_res = edr_ttp_res[doublons]
print(df_doublons_res)

Empty DataFrame
Columns: [distrcode, distrlib, distrsiret, cddept, inseecommune, nomcommune, cdreseau, nomreseau, codetypeinstallation, nomtypeinstallation, inae, cdpointsurv, nompointsurv, referenceprel, dateprel, heureprel, finaliteprel, conclusionprel, cdtypeeau, libtypeeau, plvconformitebacterio, plvconformitechimique, plvcomplet, plvcdorganismepayeur, typevisite, plvcondition, preleveur, distrsiren, plvconformiterefbacterio, plvconformiterefchimique, cddeptsandre, dtautorisation, champcaptant, codebrgm, codebss, dmajplv, dmodifplv, dmodifins, representativite, coord_x, coord_y]
Index: []
Empty DataFrame
Columns: [cddept, referenceprel, cdanalyse, refechantillonlabo, datereceptionechant, cdlaboratoire, laboratoire, cdtypeanalyse, analysecdorganisme, analysedate, insituana, cdparametre, cdparametresiseeaux, rqana, rssigne, rsana, cdunitereferencesiseeaux, cdunitereference, cdfractionanalyse, sandrelaboratoire, representativite]
Index: []


In [8]:
# création d'une liste unique de referenceprel

edr_plv = edr_ttp_plv[["referenceprel"]].copy()
unique_edr_ttp_plv = edr_plv.drop_duplicates()
unique_edr_ttp_plv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6593 entries, 0 to 6592
Data columns (total 1 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   referenceprel  6593 non-null   object
dtypes: object(1)
memory usage: 51.6+ KB


## TRAVAIL SUR EAU DE COMMUNE

In [9]:
import duckdb
from pipelines.tasks._common import DUCKDB_FILE

con = duckdb.connect(database=DUCKDB_FILE, read_only=True)

In [10]:
query_2024 = """
SELECT * from edc_prelevements
"""

edc_plv = con.sql(query_2024)
data_edc_plv = edc_plv.df()
data_edc_plv.head(2)

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont      nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                           conclusionprel                         ugelib                        distrlib                          moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    001  001000003             01007        AMBRONAY          None                None          None   00100119766 2020-02-13     11h40  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2020        2025-02-14     20230811-150005
1    001  001000003             01007        AMBRONAY     001001304  TTP (CLG) AMBRONAY         100 %   00100120290 2020-03-17     11h15  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2020        2025-02-14     20230811-150005

In [11]:
edc_plv = data_edc_plv[["referenceprel"]].copy()
unique_edc_plv = edc_plv.drop_duplicates()
unique_edc_plv.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1436114 entries, 0 to 2083344
Data columns (total 1 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   referenceprel  1436114 non-null  object
dtypes: object(1)
memory usage: 21.9+ MB


In [12]:
# extraction d'une referenceprel dnas bdd "prélèvements"

edr_ttp_plv.loc[edr_ttp_plv["referenceprel"] == "00100143348", :]

distrcode                                                   distrlib      distrsiret cddept inseecommune         nomcommune   cdreseau                           nomreseau codetypeinstallation               nomtypeinstallation inae    cdpointsurv                       nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
0  00101028  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  25010183900059    001        01004  AMBERIEU-EN-BUGEY  001000245  TTP (CLG) AMBERIEU VILLE VAREILLES                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0010000000252  TTP (CLG) AMBERIEU VILLE VAREILLE   00100143348  2024-11-25     09h02           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00100452  P1                 INF  13000807100107  250101839                        C                        C             1             NaN           NaN       NaN      NaN  2024-12-17  2024-12-18  2025-01-06                O      NaN      NaN

In [13]:
exple = """
SELECT * from edc_prelevements
WHERE referenceprel='00100143348'
"""

edc_plv_exple = con.sql(exple)
data_edc_plv_exple = edc_plv_exple.df()
data_edc_plv_exple

cddept   cdreseau inseecommuneprinc    nomcommuneprinc cdreseauamont                      nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                           conclusionprel                             ugelib                                                   distrlib                                                     moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    001  001000248             01004  AMBERIEU-EN-BUGEY     001000245  TTP (CLG) AMBERIEU VILLE VAREILLES         100 %   00100143348 2024-11-25     09h02  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SYND. EAUX REGION D'AMBERIEU-EN-B  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2024        2025-02-14     20250203-075802

In [14]:
# extraction d'un prélèvement dans bdd resultats

edr_ttp_res.loc[edr_ttp_res["referenceprel"] == "00100143348", :]

cddept referenceprel    cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux   rqana rssigne  rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
6106      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       1161.0              12DCLE   <0,10       <    0.0                     µg/L              133                NaN              217.0                O
6107      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       1141.0                 24D  <0,020       <    0.0                     µg/L              133                NaN              217.0                O
6108      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       2011.0               26DCB  <0,005       <    0.0                     µg/L              133                NaN              217.0                O
6109      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       8865.0             471811R  <0,020       <    0.0                     µg/L              133                NaN              217.0                O
6110      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       1832.0                 A2H  <0,020       <    0.0                     µg/L              133                NaN              217.0                O
...     ...           ...          ...                ...                 ...           ...           ...           ...                ...         ...       ...          ...                 ...     ...     ...    ...                      ...              ...                ...                ...              ...
6451      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       2096.0             TRINEXA  <0,020       <    0.0                     µg/L              133                NaN              217.0                O
6452      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       7087.0               TRITO  <0,020       <    0.0                     µg/L              133                NaN              217.0                O
6453      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       1295.0             TURBNFU     0,3     NaN    0.3                      NFU              232                NaN              217.0                O
6454      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       7521.0              ZCYINE  <0,005       <    0.0                     µg/L              133                NaN              217.0                O
6455      1   00100143348  100152387.0      LSE2411-32339          2024-11-25        6901.0  4.105453e+13      0011P2BC           100452.0  2024-12-11         L       2858.0             ZOXAMID  <0,005       <    0.0                     µg/L              133                NaN              217.0                O

[350 rows x 21 columns]

In [15]:
exple_res = """
SELECT * from edc_resultats
WHERE referenceprel='00100143348'
"""

edc_res_exple = con.sql(exple_res)
data_edc_res_exple = edc_res_exple.df()
data_edc_res_exple

cddept referenceprel cdparametresiseeaux  cdparametre                libmajparametre                libminparametre libwebparametre qualitparam insituana   rqana cdunitereferencesiseeaux cdunitereference  limitequal  refqual  valtraduite     casparam referenceanl  de_partition de_ingestion_date de_dataset_datetime
0      001   00100143348              12DCLE         1161             DICHLOROÉTHANE-1,2             Dichloroéthane-1,2            None           N         L   <0,10                     µg/L              133    <=3 µg/L     None          0.0     107-06-2  00100152387          2024        2025-02-14     20250203-075802
1      001   00100143348                 24D         1141                          2,4-D                          2,4-D            None           N         L  <0,020                     µg/L              133  <=0,1 µg/L     None          0.0      94-75-7  00100152387          2024        2025-02-14     20250203-075802
2      001   00100143348               26DCB         2011          2,6 DICHLOROBENZAMIDE          2,6 Dichlorobenzamide            None           N         L  <0,005                     µg/L              133  <=0,1 µg/L     None          0.0    2008-58-4  00100152387          2024        2025-02-14     20250203-075802
3      001   00100143348             471811R         8865         CHLOROTHALONIL R471811         Chlorothalonil R471811            None           N         L  <0,020                     µg/L              133        None     None          0.0         None  00100152387          2024        2025-02-14     20250203-075802
4      001   00100143348                 A2H         1832             ATRAZINE-2-HYDROXY             Atrazine-2-hydroxy            None           N         L  <0,020                     µg/L              133  <=0,1 µg/L     None          0.0    2163-68-0  00100152387          2024        2025-02-14     20250203-075802
..     ...           ...                 ...          ...                            ...                            ...             ...         ...       ...     ...                      ...              ...         ...      ...          ...          ...          ...           ...               ...                 ...
345    001   00100143348             TRINEXA         2096               TRINÉXAPAC-ÉTHYL               Trinéxapac-éthyl            None           N         L  <0,020                     µg/L              133  <=0,1 µg/L     None          0.0   95266-40-3  00100152387          2024        2025-02-14     20250203-075802
346    001   00100143348               TRITO         7087                  TRITOSULFURON                  Tritosulfuron            None           N         L  <0,020                     µg/L              133  <=0,1 µg/L     None          0.0  142469-14-5  00100152387          2024        2025-02-14     20250203-075802
347    001   00100143348             TURBNFU         1295  TURBIDITÉ NÉPHÉLOMÉTRIQUE NFU  Turbidité néphélométrique NFU            None           N         L     0,3                      NFU              232        None  <=2 NFU          0.3         None  00100152387          2024        2025-02-14     20250203-075802
348    001   00100143348              ZCYINE         7521              ZETACYPERMETHRINE              Zetacypermethrine            None           N         L  <0,005                     µg/L              133  <=0,1 µg/L     None          0.0   97955-44-7  00100152387          2024        2025-02-14     20250203-075802
349    001   00100143348             ZOXAMID         2858                       ZOXAMIDE                       Zoxamide            None           N         L  <0,005                     µg/L              133  <=0,1 µg/L     None          0.0  156052-68-5  00100152387          2024        2025-02-14     20250203-075802

[350 rows x 20 columns]

In [16]:
# recherche des prélèvements présents dans EDR et absents dans EDC

resultat = unique_edr_ttp_plv[
    ~unique_edr_ttp_plv["referenceprel"].isin(unique_edc_plv["referenceprel"])
]
print(resultat)

     referenceprel
222    00200202580
224    00200202590
225    00200202591
310    00500066210
387    00600256826
...            ...
6358   09500182117
6411   97100135799
6412   97100136148
6480   97400140467
6515   97400140882

[111 rows x 1 columns]


In [17]:
# extraction des 111 prélèvements absents de EDC

res = resultat.merge(
    edr_ttp_plv, left_on="referenceprel", right_on="referenceprel", how="inner"
)
res.describe(include="all")

referenceprel distrcode  distrlib      distrsiret cddept inseecommune nomcommune   cdreseau        nomreseau codetypeinstallation               nomtypeinstallation inae    cdpointsurv                nompointsurv    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                         libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
count            111       111       111              67    111          111        111        111              111                  111                               111  111            111                         111         111       111          111                                                                                                      111       111                                111                   111                   111        111                  111        111           16              49         67                      111                      111    111.000000             0.0           0.0       0.0      0.0         111         111         111              111      0.0      0.0
unique           111        66        65              31     42           79         79         82               82                    1                                 1    3             85                          85          42        90            4                                                                                                       34         9                                  9                     3                     3          1                   69         10            4              18         27                        3                        3           NaN             NaN           NaN       NaN      NaN          31          31          67                1      NaN      NaN
top      00200202580  01700013  R.E.S.E.  33937998401008    017        19005   ALLASSAC  017003482  USINE DE BARZAN                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0170000001589  SORTIE USINE DE TRAITEMENT  2024-12-18     10h21           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T1  ESO A TURB. < 2 SORTIE PRODUCTION                     C                     C          O             01700013  P1                 PNF  13000796600083  572025526                        C                        C           NaN             NaN           NaN       NaN      NaN  2024-12-16  2024-12-17  2024-03-11                O      NaN      NaN
freq               1         7         7               8     14            6          6          4                4                  111                               111   85              4                           4           9         3          102                                                                                                       52        39                                 39                    77                    97        111                    7         58            8               8         12                       77                       83           NaN             NaN           NaN       NaN      NaN          10          10           5              111      NaN      NaN
mean             NaN       NaN       NaN             NaN    NaN          NaN        NaN        NaN              NaN                  NaN                               NaN  NaN            NaN                         NaN         NaN       NaN          NaN                                                                                                      NaN       NaN                                NaN                   NaN       

In [25]:
res.head(10)

referenceprel distrcode                             distrlib      distrsiret cddept inseecommune           nomcommune   cdreseau                        nomreseau codetypeinstallation               nomtypeinstallation inae    cdpointsurv                  nompointsurv    dateprel heureprel finaliteprel                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     conclusionprel cdtypeeau                          libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
0   00200202580  00299999                    ORGANISME INCONNU             NaN    002        59461           PETIT-FAYT  002004232               PETIT FAYT POMPAGE                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0020000004023            PETIT FAYT POMPAGE  2024-12-10     10h21           CS                                                                                                                                                                                                                                                                                                                                                                                                                      Eau d'alimentation conforme aux limites de qualité et non conforme aux références de qualité.        T1   ESO A TURB. < 2 SORTIE PRODUCTION                     C                     C          O             00299999  P1                 NaN             NaN        NaN                        C                        N             2             NaN           NaN       NaN      NaN  2025-01-24  2025-01-27  2024-03-11                O      NaN      NaN
1   00200202590  00299999                    ORGANISME INCONNU             NaN    002        80410                  HAM  002004225                  TTP MIRROIR HAM                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0020000004016       MIRROIR HAM POUR PITHON  2024-12-09     10h32           CS                                                                                                                                                                                                                                                                                                                                                                                                                      Eau d'alimentation non-conforme aux limites de qualité et conforme aux références de qualité.        T           EAU DISTRIBUEE DESINFECTEE                     S                     N          O             00299999  P1                 NaN             NaN        NaN                        S                        C             2             NaN           NaN       NaN      NaN  2025-01-24  2025-01-27  2024-03-11                O      NaN      NaN
2   00200202591  00299999                    ORGANISME INCONNU             NaN    002        80410                  HAM  002004225                  TTP MIRROIR HAM                  TTP  STATION DE TRAITEMENT-PRODUCTION  ESO  0020000004016       MIRROIR HAM POUR PITHON  2024-12-09     10h30           CS                                                                                                                                                                                                                           

In [26]:
exple = """
SELECT * from edc_prelevements
WHERE inseecommuneprinc='15141' and dateprel='2024-12-10'
"""

edc_plv_exple = con.sql(exple)
data_edc_plv_exple = edc_plv_exple.df()
data_edc_plv_exple

cddept   cdreseau inseecommuneprinc      nomcommuneprinc cdreseauamont           nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                                                                                    conclusionprel                            ugelib                       distrlib                         moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    015  015000350             15141  NEUSSARGUES-MOISSAC     015006301  NEUSSARGUES RIVE GAUCHE           5 %   01500120983 2024-12-10     09h57  Résultats d'analyses conformes à la norme. Toutefois, taux de chlore é levé et présence excessive de germes à 36°C : eau de qualité sanitaire non satisfaisante.  COMMUNE DE NEUSSARGUES - MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC                     C                     C                          C                          C          2024        2025-02-14     20250203-075802
1    015  015000359             15141  NEUSSARGUES-MOISSAC          None                     None          None   01500120979 2024-12-10     09h18                                                                                          Eau de qualité sanitaire satisfaisante. Cependant, taux de chlore élevé.  COMMUNE DE NEUSSARGUES - MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC                     C                     C                          C                          C          2024        2025-02-14     20250203-075802
2    015  015006301             15141  NEUSSARGUES-MOISSAC          None                     None          None   01500120983 2024-12-10     09h57  Résultats d'analyses conformes à la norme. Toutefois, taux de chlore é levé et présence excessive de germes à 36°C : eau de qualité sanitaire non satisfaisante.  COMMUNE DE NEUSSARGUES - MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC                     C                     C                          C                          C          2024        2025-02-14     20250203-075802
3    015  015006301             15141  NEUSSARGUES-MOISSAC     015006302       TTP CHLORE CHEYLAT         100 %   01500120984 2024-12-10     09h42                                             Résultats d'analyses conformes à la norme. Toutefois, couleur excessive : eau de qualité sanitaire non satisfaisante.  COMMUNE DE NEUSSARGUES - MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC  MAIRIE DE NEUSSARGUES-MOISSAC                     C                     C                          C                          N          2024        2025-02-14     20250203-075802

CONCLUSION : 

La base EDR TTP 2024 contient 6593 prélèvements
111 de ces prélèvements sont absents de la base EDC (soit 1.7%). 
Des vérifications on été faites sur inseecommune/date/heure pour qq prélèvements pour pallier à une anomalie sur la référence. 

Aucune correspondance n'a été détectée. 

Réfléchir sur l'intérêt d'étendre ces résultats sur la totalité des années 

Réfléchir sur l'intérêt d'intégrer ces informations dans la base EDC